app/ui-react/syndesis/src/modules/data/pages/VirtualizationMetricsPage.tsx

```diff
@@ -25,54 +27,74 @@ export const VirtualizationMetricsPage: React.FunctionComponent = () => {
    */
   const { model: virtualization } = useVirtualization(params.virtualizationId);
 
+  /**
+   * Hook to obtain view descriptors. Also does polling to get any view descriptor updates.
+   */
+  const {
+    error,
+    hasData: hasViewDefinitionDescriptors,
+    model: viewDefinitionDescriptors,
+  } = useViewDefinitionDescriptors(params.virtualizationId);
```

Comment:
Just doesn't *feel* right to me to have the metrics page fetch view information since it doesn't use that information. It is just being passed through. I'm wondering if you just move the `useViewDefinitionDescriptors` to the `VirtualizationEditorPage`. Then you wouldn't need the `haveValidView` property. And wouldn't need the `WithLoader` below either. Wdyt @indraraj @mdrillin ?

Reply:
After thinking about this more, I requested a change to the backend - https://issues.jboss.org/browse/TEIIDTOOLS-866 .  I think this should be a pretty straightforward change, and would eliminate some cases of us needing to make the extra call to get viewDefinitionDescriptors.